src/services/EdFi.DataManagementService.Api/Modules/DiscoveryModule.cs

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.DataManagementService.Api.Configuration;
+using EdFi.DataManagementService.Api.Content;
+using EdFi.DataManagementService.Api.Infrastructure.Extensions;
+using Microsoft.Extensions.Options;
+
+namespace EdFi.DataManagementService.Api.Modules;
+
+public class DiscoveryModule : IModule
+{
+    public void MapEndpoints(IEndpointRouteBuilder endpoints)
+    {
+        endpoints.MapGet("/", GetApiDetails);
+    }
+
+    internal async Task GetApiDetails(
+        HttpContext httpContext,
+        IVersionProvider versionProvider,
+        IDataModelProvider dataModelProvider,
+        IOptions<AppSettings> appSettings
+    )
+    {
+        var dataModels = dataModelProvider.GetDataModels().ToArray();
+
+        var result = new DiscoveryApiDetails(
+            version: versionProvider.Version,
+            suite: versionProvider.Suite,
+            dataModels,
+            GetUrlsByName()
+        );
+
+        await httpContext.Response.WriteAsSerializedJsonAsync(result);
+
+        Dictionary<string, string> GetUrlsByName()
+        {
+            var urlsByName = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var rootUrl = httpContext.Request.RootUrl();
+            urlsByName["dependencies"] = $"{rootUrl}/metadata/data/dependencies";
+            urlsByName["openApiMetadata"] = $"{rootUrl}/metadata";
+            urlsByName["oauth"] = appSettings.Value.AuthenticationService ?? string.Empty;
+            urlsByName["dataManagementApi"] = $"{rootUrl}/data/";
```

Comment:
Please remove "/data" here.

Or, alternately, could change the "catch all" mapping for Resources and Descriptor API to use `/data` in the route. Either way is fine with me right now.

Reply:
Will update to data/v3 and update  Resources and Descriptor API to use /data/v3